### PR TITLE
Added filtering functionality to the `list` and `help` commands

### PR DESCRIPTION
**Pull request by naceto**

### CODE REVIEW

**Review by naps250 — APPROVED** [review]

File: SteakBot.Core/Modules/HelpModule.cs [hunk 1]
```diff
@@ -24,9 +27,56 @@ public HelpModule(MemeService memeService, CommandService commandService)
...
+        [Summary("Filters and lists commands with their summaries")]
+        [Remarks("Usage: `help <filter(text)>`")]
+        public async Task Help(string filter)
+        {
+            var commands = _commandService.Commands;
+            if (!string.IsNullOrWhiteSpace(filter))
+            {
+                commands = commands
+                    .Where(x => x.Name.Contains(filter) ||
+                                (x.Remarks != null && x.Remarks.Contains(filter)));
```
Comment (penev92) [inline SteakBot.Core/Modules/HelpModule.cs]: This should also check the commands' Summary.

**Review by penev92 — CHANGES_REQUESTED** [review]
Looks good with one small request.

File: SteakBot.Core/Modules/HelpModule.cs (hunk 1 above)
Comment (naceto) [reply SteakBot.Core/Modules/HelpModule.cs]: Done.

**Comment by penev92** [comment]
Looks good to me now 👍

**Review by penev92 — APPROVED** [review]
Looks good to me now 👍

**Pull request merged**